backend/src/features/game-identifier/game.identifier.js

```diff
@@ -144,26 +144,44 @@ export class GameIdentifier {
       return;
     }
 
-    const htmlDetailsPages = await this.#getSteamDbHtmlDetailsPages(games);
+    const steamApps = await this.#steamAppsRepository.getSteamAppsById(games);
+
+    const [htmlDetailsPages, appsWithoutPages] = await this.#getSteamDbHtmlDetailsPages(
+      games,
+      steamApps,
+    );
 
     const updatedGames = updateMissingDetails(games, htmlDetailsPages);
 
-    this.#persistMissingProperties(updatedGames);
+    this.#persistMissingProperties(updatedGames, appsWithoutPages);
   };
 
-  async #getSteamDbHtmlDetailsPages(games) {
-    const htmlDetailsPages = [];
+  async #getSteamDbHtmlDetailsPages(games, steamApps) {
```

Comment:
from looking at the method signature and the code its not obvious that it can be used without the second argument like you do in line 205

you should break this method into two methods:
- m1: `getSteamDbHtmlDetailsPages(games)` gets the htmlPages and handles the error, makes an empty entry for all empty pages, returns the `htmlDetailsPages`, basically like before
- m2: `recordFailedAttemts(htmlDetailsPages, steamApps)` walks through the `htmlDetailsPages` and creates `appsWithoutPages` like in the if statement line 166 and returns it

now you can call the first method also in line 205